PythonVisualizations/Heap.py

```diff
@@ -0,0 +1,518 @@
+        half_cell_y = (self.CELL_HEIGHT - self.CELL_BORDER) // 2
+
+        return add_vector(subtract_vector(self.cellCoords(index), (0, half_cell_y)), (half_cell_x, 0))
+
+
+    def createArrayCell(self, index):  # Create a box representing an array cell
+        cell_coords = self.cellCoords(index)
+        half_border = self.CELL_BORDER // 2
+        arrayCoords = add_vector(cell_coords,
+                                (half_border - 1, half_border - 1,
+                                self.CELL_BORDER - half_border - 1, self.CELL_BORDER - half_border))
+        rect = self.canvas.create_rectangle(arrayCoords,
+                                            fill=None, outline=self.CELL_BORDER_COLOR, width=self.CELL_BORDER,
+                                            tags="arrayCell")
+        self.canvas.lower(rect)
+        return rect
+
+
+    def createCellValue(self, indexOrCoords, key, color=None):
+        """Create new canvas items to represent a cell value.  A square
+        is created filled with a particular color with an text key centered
+        inside.  The position of the cell can either be an integer index in
+        the Array or the bounding box coordinates of the square.  If color
+        is not supplied, the next color in the palette is used.
+        An event handler is set up to update the VisualizationApp argument
+        with the cell's value if clicked with any button.
+        Returns the tuple, (square, text), of canvas items
+        """
+        # Determine position and color of cell
+        if isinstance(indexOrCoords, int):
+            rectPos = self.cellCoords(indexOrCoords)
+            valPos = self.cellCenter(indexOrCoords)
+        else:
+            rectPos = indexOrCoords
+            valPos = divide_vector(add_vector(rectPos[:2], rectPos[2:]), 2)
+        if color is None:
+            # Take the next color from the palette
+            color = drawable.palette[Heap.nextColor]
+            Heap.nextColor = (Heap.nextColor + 1) % len(drawable.palette)
+        cell_rect = self.canvas.create_rectangle(
+            *rectPos, fill=color, outline='', width=0)
+        cell_val = self.canvas.create_text(
+            *valPos, text=str(key), font=self.SMALL_FONT, fill=self.VALUE_COLOR)
+        handler = lambda e: self.setArgument(str(key))
+        for item in (cell_rect, cell_val):
+            self.canvas.tag_bind(item, '<Button>', handler)
+        return cell_rect, cell_val
+
+
+    def display(self):
+        self.canvas.delete("all")
+        for i in range(self.size):  # Draw grid of cells
+            self.createArrayCell(i)
+        # go through each Drawable in the list
+        for i, n in enumerate(self.list):
+            # create display objects for the associated Drawables
+            n.display_shape, n.display_val = self.createCellValue(
+                i, n.val, n.color)
+            n.color = self.canvas.itemconfigure(n.display_shape, 'fill')

```

Comment:
The `canvas.itemconfigure()` method returns a tuple.  The last element of the tuple is the value of the configuration parameter, so you need a `[-1]` after the call to extract the value.